### PR TITLE
Clarify module descriptions in rst AND delete unused file path checks, context managers, and input type checks in spatial.py

### PR DESCRIPTION
**Pull request by mgraber**
This pull request addresses issue #67 

Unused unit tests and commented code lines are deleted from spatial.py
The file get-started.rst now includes descriptions of each of the three modules and converts them to list form.

### CODE REVIEW

**Comment by lwasser** [comment]
@mgraber this PR was first submitted before I gave you the go ahead! that typically is frowned upon when dealing with developers in other repos and so please follow instructions in the future.

Also this PR is failing tests because the `.rst` format is not correct. Please review the .rst lesson carefully to ensure your syntax is correct. It is important that you follow syntax and guidelines when trying to contribute to packages.

File: docs/get-started.rst [hunk 1]
```diff
@@ -8,9 +8,19 @@ lab earth analytics courses.
 Earthpy Modules
 ---------------
 
-Spatial
-IO
-Utils
+- Spatial: Various functions for loading, manipulating, and displaying raster data
+  - `create_raster_stack`: generates a single multi-band raster from separate raster TIFs,
```
Comment (lwasser) [inline docs/get-started.rst]: Did you intent this to be `create_raster_stack` using code highlighting?

File: docs/get-started.rst [hunk 2]
```diff
@@ -8,9 +8,19 @@ lab earth analytics courses.
...
+  such as a collection of single-band TIFs from Landsat
+  - `normalized_diff`: calculate a normalized difference for two bands of a multi-band raster
+  - `hillshade`: create a hillshade array from image elevation data
+  - `crop_image`: crop raster data to a polygon
+  - Custom plotting shortcuts, such as `colorbar` for appropriately sized legends, `create_legend` for
+  legends showing a color for each class of a raster, `plot_bands` for creating a grid of images
+  showing each individual band of a raster, and `plot_rgb` for quickly generating
+  composite images or false color images from raster bands.
+- IO: Handles data search and retrieval from EarthLab, supporting analytics courses
+available here: https://www.earthdatascience.org/courses/
```
Comment (lwasser) [inline docs/get-started.rst]: Please look up the .rst syntax for urls and fix this link.  Please read the lessons before updating this PR @mgraber 

i'm actually not sure which line is failing travis tests but i wonder if it is this one. 
https://travis-ci.org/earthlab/earthpy/builds/446396349?utm_source=github_status&utm_medium=notification

```
get-started.rst:22:Bullet list ends without a blank line; unexpected unindent.
```

as the url is not formatted correctly but als i'm not sure if that last line is supposed to be a part of the previous bullet. if it is i think it needs to be indented.

File: docs/get-started.rst (hunk 1 above)
Comment (mgraber) [reply docs/get-started.rst]: Sorry. I'd thought that having input from you and Max on what the changes should include was my go ahead. Lesson learned.

**Comment by mgraber** [comment]
@lwasser Made some updates to try to fix the bullet indentation and formatting. It passes the Travis CI tests now.

**Comment by lwasser** [comment]
Wonderful. Thank you @mgraber !! I am merging this. Congratulations on your first open source contribution. Your edits should be live on earthpy.readthedocs.io within the next hour. The build happens on readthedocs once i merge! Be sure to have a look to ensure things look like you think they should.

**Pull request merged**